index.js

```diff
@@ -22,31 +23,26 @@ module.exports = function(opts) {
 
     var filepath = module.exports.getTplPath(req, opts);
 
-    // Handle only .jade files
-    if (opts.ext.indexOf(path.extname(filepath)) === -1)
-      return next();
+    if (!filepath)
+      return res.send(404)
 
     if (filepath.indexOf(opts.baseDir) !== 0)
-      return next(new Error('Invalid path'));
+      return res.send(403);
```

Comment:
Yeah.. I'm a bit lazy :) Not sure if we able to even test.. I guess `url.parse` excapes rubbish like `..` etc.